pkg/roachpb/metadata_test.go

```diff
@@ -186,6 +186,34 @@ func TestLocalityConversions(t *testing.T) {
 	}
 }
 
+func TestLocalityMatches(t *testing.T) {
```

Comment:
Potential cases to add here:

- Empty filter
- Empty Tiers on the receiving locality

Reply:
Done